### ISSUE TITLE
Error when loading an empty shapefile

### ISSUE THREAD

**Issue by WeatherGod** [geopandas/geopandas#649]
as of v0.3.0, when loading a valid shapefile that has no records, an error is raised:
```
import geopandas as gpd
df = gpd.read_file('active_perimeters_dd83.shp')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/broot/scratch/miniconda/lib/python2.7/site-packages/geopandas/io/file.py", line 30, in read_file
    gdf = gdf[columns]
  File "/home/broot/scratch/miniconda/lib/python2.7/site-packages/geopandas/geodataframe.py", line 398, in __getitem__
    result = super(GeoDataFrame, self).__getitem__(key)
  File "/home/broot/scratch/miniconda/lib/python2.7/site-packages/pandas/core/frame.py", line 2133, in __getitem__
    return self._getitem_array(key)
  File "/home/broot/scratch/miniconda/lib/python2.7/site-packages/pandas/core/frame.py", line 2177, in _getitem_array
    indexer = self.loc._convert_to_indexer(key, axis=1)
  File "/home/broot/scratch/miniconda/lib/python2.7/site-packages/pandas/core/indexing.py", line 1269, in _convert_to_indexer
    .format(mask=objarr[mask]))
KeyError: "[u'IRWINID' u'MAPMETHOD' u'UNITIDPROT' u'UNITIDOWN' u'INCIDENTID'\n u'FIRENAME' u'PERDATTIME' u'COMMENTS' u'AGENCY' u'ACTIVE' u'COMPPARID'\n u'FIREYEAR' u'DATECRNT' u'INCIWEBID' u'FIRECODE' u'COMPFIRECD' u'FIRENUM'\n u'COMPLEXNM' u'STATE' u'INCOMPLEX' u'GISACRES' u'MERGEID' 'geometry'] not in index"
```

Empty shapefiles are perfectly valid in my processing, as the shapefile is ingested daily to process the polygons of any active wildfires. At the moment, there are no active wildfires in the US with polygons defined.

In v0.2.1, there was a different, but still undesirable behavior:
```
>>> import geopandas as gpd
>>> df = gpd.read_file('active_perimeters_dd83.shp')
>>> df
Empty GeoDataFrame
Columns: []
Index: []
```
The shapefile has columns defined, and fiona can see those columns just fine, too. But geopandas just silently ignores all of that. This is undesirable because one of my processing steps is to validate the schema of the input shapefile. If there is no columns, then I don't know if that is because there is no data, or because the wrong input file was selected.
[active_perimeters_dd83.zip](https://github.com/geopandas/geopandas/files/1665094/active_perimeters_dd83.zip)

**Comment by jorisvandenbossche** [comment]
@WeatherGod Thanks for the report! 
That should be an easy fix, see https://github.com/geopandas/geopandas/pull/653. This will now return an empty geodataframe with the correct columns set.

**Comment by HenryW95** [comment]
On v0.3.0 read_file on an empty geojson results in `KeyError: "['geometry'] not in index"` but seems to be working fine on v0.2.1

**Comment by kjordahl** [comment]
@HenryW95 I can replicate this. Could you please make a new issue instead of piggybacking on this one?

**Comment by jorisvandenbossche** [comment]
This seems to be working again on master (it's similar to this issue, which was closed by PR  #653):

```
In [36]: import geopandas

In [37]: geopandas.__version__
Out[37]: '0.3.0+87.g8706b7d.dirty'

In [38]: empty_geojson = '{"type": "FeatureCollection", "features": []}'

In [39]: with open('__empty.geojson', 'w') as f:
    ...:     f.write(empty_geojson)
    ...:     
    ...:     

In [40]: geopandas.read_file('__empty.geojson')
Out[40]: 
Empty GeoDataFrame
Columns: [geometry]
Index: []
```
